pkg/operator/ceph/controller/cluster_info.go

```diff
@@ -124,9 +130,32 @@ func CreateOrLoadClusterInfo(clusterdContext *clusterd.Context, context context.
 			MonitorSecret: string(secrets.Data[MonSecretNameKey]),
 			Context:       context,
 		}
+
 		if cephUsername, ok := secrets.Data[CephUsernameKey]; ok {
 			clusterInfo.CephCred.Username = string(cephUsername)
 			clusterInfo.CephCred.Secret = string(secrets.Data[CephUserSecretKey])
+			// for upgrade cluster get rookoperator keyrings
+			if string(cephUsername) == "client.admin" {
```

Comment:
We need to generate the ceph connection config on disk with the existing keyring before we can try to generate the new keyring. Try this:
```go
if string(cephUsername) == "client.admin" {

	// write the existing admin keyring to the config dir
	if _, err := cephclient.GenerateConnectionConfig(clusterdContext, clusterInfo); err != nil {
		return nil, maxMonID, monMapping, errors.Wrap(err, "failed to write connection config for existing admin keyring")
	}

	// generate the new admin keyring
	userKey, err := client.AuthGetOrCreateKey(clusterdContext, clusterInfo, cephclient.AdminUsername, adminCapArgs)
	if err != nil {
		return nil, maxMonID, monMapping, errors.Wrap(err, "failed to create/update rookoperator keyring")
	}
	...

```

Reply:
done